src/hotspot/share/cds/archiveHeapWriter.cpp

```diff
@@ -565,6 +634,12 @@ bool ArchiveHeapWriter::is_marked_as_native_pointer(ArchiveHeapInfo* heap_info,
   assert((Metadata**)_requested_bottom <= requested_field_addr && requested_field_addr < (Metadata**) _requested_top, "range check");
 
   BitMap::idx_t idx = requested_field_addr - (Metadata**) _requested_bottom;
+  // Leading zeros have been removed so some addresses may not be in the ptrmap
+  if (idx < MetaspaceShared::ptrmap_start_pos()) {
```

Comment:
This is now the only place you need to know `ptrmap_start_pos` during dump time (at runtime you can get the info from `FileMapInfo::current_info()`). I think it's better to store `ptrmap_start_pos` as a static field in the `ArchiveHeapWriter` class instead.